src/Lean/Widget/InteractiveDiagnostic.lean

```diff
@@ -96,15 +96,10 @@ where
 
   go : NamingContext → Option MessageDataContext → MessageData → MsgFmtM Format
   | _,    _,         ofFormat fmt             => withIgnoreTags (pure fmt)
-  | _,    _,         ofLevel u                => return format u
-  | _,    _,         ofName n                 => return format n
-  | nCtx, some ctx,  ofSyntax s               => withIgnoreTags (ppTerm (mkPPContext nCtx ctx) ⟨s⟩) -- HACK: might not be a term
```

Comment:
This change looks a bit fishy since `withIgnoreTags` is no longer used.

Reply:
Okay, I convinced myself that this is fine, but man this code is awfully complicated lol. I wrote a docstring trying to explain what's going on and pushed it directly here. Hope that's okay, but do let me know if you'd rather I not push directly.